### PR TITLE
Remove Brave Talk feature flag (used on NTP)

### PR DESCRIPTION
**Pull request by bsclifton**
Fixes https://github.com/brave/brave-browser/issues/18426

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used GitHub [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
See https://github.com/brave/brave-browser/issues/18426

### CODE REVIEW

File: browser/extensions/api/brave_talk_api.cc [hunk 1]
```diff
@@ -23,8 +21,7 @@ ExtensionFunction::ResponseAction BraveTalkIsSupportedFunction::Run() {
     return RespondNow(Error("Not available in Tor profile"));
   }
 
-  bool is_supported =
-      base::FeatureList::IsEnabled(brave_talk::features::kBraveTalk);
+  bool is_supported = true;
```
Comment (simonhong) [inline browser/extensions/api/brave_talk_api.cc:24]: nit: how about  directly like `return RespondNow(OneArgument(base::Value(true)));`?
Or, if it's for documentation purpose, constant(`constexpr`) would be better?

**Review by simonhong — APPROVED** [review]
++

File: browser/extensions/api/brave_talk_api.cc (hunk 1 above)
Comment (bsclifton) [reply browser/extensions/api/brave_talk_api.cc:24]: Good call! I'll mark PR as CI/skip and push a fix

**Comment by bsclifton** [comment]
Last CI ran passed on all platforms; marked as `CI/skip` and pushed a small fix (see comment)

**Comment by stephendonner** [comment]
### Verified `PASSED` using 

Brave | 1.32.46 Chromium: 94.0.4606.71 (Official Build) nightly (x86_64)
-- | --
Revision | 1d32b169326531e600d836bd395efc1b53d0f6ef-refs/branch-heads/4606@{#1256}
OS | macOS Version 11.6 (Build 20G165)

### Steps:
1. new profile
2. launched Brave using `--enable-logging=stderr --vmodule="*/variations/*"=6,"*/bat-native-ledger/*"=6,"*/brave_rewards/*"=6,"*/bat-native-ads/*"=6,"*/bat-native-confirmations/*"=6,"*/brave_ads/*"=9,"*/brave_user_model/*"=6 --brave-ads-staging --brave-ads-debug --rewards=staging=true,reconcile-interval=3 --variations-server-url=https://variations.bravesoftware.com/seed`
3. opened a new-tab page
4. confirmed and tested the Brave Talk video-camera icon launches `https://talk.brave.com/widget`
5. confirmed Brave Talk's widget shows up in `Edit Cards` -> `Cards` -> `+ Add`
6. confirmed clicking on `+ Add` adds `Brave Talk` to the the active widgets on the NTP
7. confirmed clicking on `Start video call` takes the user to `https://talk.brave.com/widget`
8. confirmed `About your data` link goes to `https://brave.com/privacy/browser/#brave-talk-learn`
9. confirmed that traveling to the future, 3 days from now, the Brave Talk promo "tooltip" popup shows up

example | example | example | example | example | example
---------|----------|---------|----------|---------|---------
Screen Shot 2021-10-06 at 10 55 48 AM | Screen Shot 2021-10-06 at 10 56 41 AM | Screen Shot 2021-10-06 at 10 56 48 AM | Screen Shot 2021-10-06 at 10 56 54 AM | Screen Shot 2021-10-06 at 10 56 58 AM | Screen Shot 2021-10-09 at 11 38 38 AM

**Pull request merged**